sitemap.php

```diff
@@ -0,0 +1,102 @@
+<?php
+# MantisBT - A PHP based bugtracking system
+
+# MantisBT is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# MantisBT is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with MantisBT.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Sitemap - http://sitemaps.org
+ *
+ * @package MantisBT
+ * @copyright Copyright 2002 MantisBT Team - mantisbt-dev@lists.sourceforge.net
+ * @link http://www.mantisbt.org
+ *
+ * @uses access_api.php
+ * @uses authentication_api.php
+ * @uses config_api.php
+ * @uses database_api.php
+ * @uses filter_api.php
+ * @uses filter_constants_inc.php
+ */
+
+# Prevent output of HTML in the content if errors occur
+define( 'DISABLE_INLINE_ERROR_REPORTING', true );
+
+require_once( 'core.php' );
+require_api( 'access_api.php' );
+require_api( 'authentication_api.php' );
+require_api( 'config_api.php' );
+require_api( 'database_api.php' );
+require_api( 'filter_api.php' );
+require_api( 'filter_constants_inc.php' );
+
+# Access level to get a sitemap
+$t_threshold = config_get( 'view_bug_threshold' );
+
+# Maximum number of URLs
+$t_max_urls = 50000;
+
+$t_path = config_get_global( 'path' );
+
+/**
+ * Print a sitemap URL.
+ *
+ * @param string $p_url The URL: has to be relative to the installation path {@see $g_path}.
+ */
+function print_sitemap_url( string $p_url ) {
+	global $t_path;
+	echo '<url>', "\n",
+		'<loc>', htmlspecialchars( $t_path . $p_url, ENT_XML1 | ENT_QUOTES ), '</loc>', "\n",
+		'</url>', "\n";
+}
+
+header( 'Content-Type: application/xml; charset=UTF-8' );
+
+echo '<?xml version="1.0" encoding="UTF-8"?>', "\n",
+	'<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">', "\n";
+
+if( auth_anonymous_enabled() ) {
```

Comment:
Should you use `current_user_is_anonymous()` instead?

Reply:
The current_user_is_anonymous() function not only uses the auth_anonymous_enabled() function, but also checks cookies in the database, which seems unnecessary in this situation.